src/main/webapp/app/tasks/task-list/task-list.component.ts

```diff
@@ -1,28 +1,62 @@
-import { ChangeDetectionStrategy, Component, EventEmitter, Inject, Input, Output } from '@angular/core';
-
-import { Task } from '../task';
-import { TaskService } from '../task.service';
-
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Inject,
+  Input,
+  Output,
+} from '@angular/core'
+import { Task } from '../task'
+import { TaskService } from '../task.service'
+import { MatCheckboxChange } from '@angular/material/checkbox'
 /**
  * A list of tiny tasks.
  */
 @Component({
   selector: 'tiny-task-list',
   templateUrl: './task-list.component.html',
   styleUrls: ['./task-list.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class TaskListComponent {
+  @Input() tasks: Task[]
+
+  @Output() deleted: EventEmitter<Task> = new EventEmitter()
 
-  @Input() tasks: Task[];
+  @Output() statusChanged: EventEmitter<Task> = new EventEmitter()
 
-  @Output() deleted: EventEmitter<Task> = new EventEmitter();
+  @Output() deletedDoneTasks: EventEmitter<Task> = new EventEmitter()
 
-  constructor(@Inject('TaskService') private taskService: TaskService) { }
+  constructor(@Inject('TaskService') private taskService: TaskService) {}
 
   delete(task: Task): void {
     this.taskService.delete(task.id).subscribe(() => {
-      this.deleted.emit(task);
-    });
+      this.deleted.emit(task)
+    })
+  }
+
+  setIsDone(task: Task): void {
+    this.taskService
+      .setIsDone(task.id)
+      .subscribe(() => this.statusChanged.emit(task))
+  }
+
+  deleteDoneTasks(task: Task): void {
```

Comment:
the task does not need to be passed to remove all status = done entries from the array later on, so I would change the siganture to  `deleteDoneTasks(task: Task): void`